### DISCUSSION TITLE
How do I go about setting the current datetime as the default value for a 'Date' field?

### DISCUSSION THREAD

**Discussion by Ashvith10** [mikro-orm/mikro-orm#6279] [Q&A] [answered]
So far, I've come across `defaultRaw`. Alternatively, there's also the `onCreate` field, which isn't the same thing, but I was wondering if this could be done only with the `default` field? Because when I try using `new Date()` or `Date.now()` with this particular 'Date' column.

**Comment by Ashvith10** [comment]
By using the [`sql`](https://mikro-orm.io/api/core/function/sql) tagged template functions, which is a part of the core library. Below is an example using the EntitySchema.

```js
import { EntitySchema, sql } from '@mikro-orm/core'

...
...

export const someSchema = new EntitySchema({
    class: someClass, 
    properties: {
        ...,
        ...,
        ...,
        date: { type: 'Date', defaultRaw: sql.now() } 
    }
})
```

**Reply by B4nan** [reply, marked as answer]
You can as well just use `default: sql.now()`, or `defaultRaw: 'current_timestamp'`, since `defaultRaw` expects an SQL string itself. `sql.now()` is a shortcut for `raw('current_timestamp')` basically.

**Reply by Ashvith10** [reply]
@B4nan I think I like the `default` syntax more, thank you for suggesting the same - not only is it three characters less, but it also maps properly with the database constraint of the same name.
